src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network.impl;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.contingency.Contingency;
+import com.powsybl.contingency.ContingencyElement;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.LoadDetail;
+import com.powsybl.openloadflow.graph.GraphDecrementalConnectivity;
+import com.powsybl.openloadflow.network.*;
+import com.powsybl.openloadflow.util.PerUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Gaël Macherel <gael.macherel@artelys.com>
+ */
+public class PropagatedContingency {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PropagatedContingency.class);
+
+    private final Contingency contingency;
+
+    private final int index;
+
+    private final Set<String> branchIdsToOpen;
+
+    private final Set<Switch> switchesToOpen;
+
+    private final Set<String> hvdcIdsToOpen;
+
+    private final Map<String, Double> generatorIdsToLose;
+
+    private final Map<String, PowerShift> loadIdsToShift;
+
+    private final Map<String, Double> shuntIdsToShift;
```

Comment:
I used shift because is only part of an aggregate

Reply:
Yes, better.